service/migration_manager.cc

```diff
@@ -857,16 +857,23 @@ future<std::vector<mutation>> prepare_new_view_announcement(storage_proxy& sp, v
 #endif
     auto& db = sp.local_db();
     try {
-        auto&& keyspace = db.find_keyspace(view->ks_name()).metadata();
+        auto keyspace = db.find_keyspace(view->ks_name()).metadata();
         if (keyspace->cf_meta_data().contains(view->cf_name())) {
             throw exceptions::already_exists_exception(view->ks_name(), view->cf_name());
         }
         mlogger.info("Create new view: {}", view);
-        auto mutations = db::schema_tables::make_create_view_mutations(keyspace, std::move(view), ts);
-        co_return co_await include_keyspace(sp, *keyspace, std::move(mutations));
+        return seastar::async([&db, keyspace = std::move(keyspace), &sp, view = std::move(view), ts] {
```

Comment:
Why are you de-cooroutinizing this function?

Reply:
Because I had to add seastar::async (I forgot to explain this in the commit message - the on_* callbacks all require to be run in a Seastar thread). Since async() returns a future, it's easier to just return this future and be done with it, no need to do the "co_return co_await" song and dance. The only other co_return in this function was an exception return, and also was only made more complicated by by the coroutinization.